src/ext/java/org/opentripplanner/ext/siri/AffectsMapper.java

```diff
@@ -91,117 +94,105 @@ private List<EntitySelector> mapVehicleJourneys(AffectsScopeStructure.VehicleJou
-              new EntitySelector.Unknown(
-                "Alert affects unknown vehicle journey %s".formatted(vehicleJourneyRef.getValue())
               )
             );
           }
 
-          for (FeedScopedId tripId : tripIds) {
-            if (!affectedStops.isEmpty()) {
-              for (AffectedStopPointStructure affectedStop : affectedStops) {
-                FeedScopedId stop = getStop(
-                  affectedStop.getStopPointRef().getValue(),
-                  feedId,
-                  transitService
-                );
-                if (stop == null) {
-                  stop = new FeedScopedId(feedId, affectedStop.getStopPointRef().getValue());
-                }
-                // Creating unique, deterministic id for the alert
-                EntitySelector.StopAndTrip entitySelector = new EntitySelector.StopAndTrip(
-                  stop,
-                  tripId,
-                  serviceDate,
-                  resolveStopConditions(affectedStop.getStopConditions())
-                );
-                selectors.add(entitySelector);
-              }
-            } else {
-              selectors.add(new EntitySelector.Trip(tripId, serviceDate));
-            }
+          if (tripIds.isEmpty()) {
+            // Neither referenced Trip nor Fuzzy-matching found Trip - add Trip-id as-is
+            tripIds.add(entityResolver.resolveId(vehicleJourneyRef.getValue()));
           }
+
+          affectedTripIds.addAll(tripIds);
         }
+
+        selectors.addAll(mapTripSelectors(affectedStops, affectedTripIds, serviceDate));
       }
 
       final FramedVehicleJourneyRefStructure framedVehicleJourneyRef = affectedVehicleJourney.getFramedVehicleJourneyRef();
       if (framedVehicleJourneyRef != null) {
-        final DataFrameRefStructure dataFrameRef = framedVehicleJourneyRef.getDataFrameRef();
-        final String datedVehicleJourneyRef = framedVehicleJourneyRef.getDatedVehicleJourneyRef();
-
-        FeedScopedId tripId = getTripId(datedVehicleJourneyRef, feedId, transitService);
+        selectors.addAll(
+          mapTripSelectors(
+            affectedStops,
+            List.of(entityResolver.resolveId(framedVehicleJourneyRef.getDatedVehicleJourneyRef())),
+            entityResolver.resolveServiceDate(framedVehicleJourneyRef)
+          )
+        );
+      }
 
-        if (tripId != null) {
-          LocalDate serviceDate = null;
-          if (dataFrameRef != null && dataFrameRef.getValue() != null) {
-            serviceDate = LocalDate.parse(dataFrameRef.getValue());
-          }
+      final List<DatedVehicleJourneyRef> datedVehicleJourneyReves = affectedVehicleJourney.getDatedVehicleJourneyReves();
+      if (isNotEmpty(datedVehicleJourneyReves)) {
+        List<FeedScopedId> tripIds = new ArrayList<>();
+        for (DatedVehicleJourneyRef datedVehicleJourneyRef : datedVehicleJourneyReves) {
+          tripIds.add(entityResolver.resolveId(datedVehicleJourneyRef.getValue()));
```

Comment:
`datedVehicleJourneyRef` is a Reference to DSJ, we should use `entityResolver.resolveTripOnServiceDate` here, instead of above.

Reply:
`datedVehicleJourneyRef` is _usually_ a DSJ in a SIRI SX context. 

However, the corner-case is when a trip is added in SIRI ET (i.e. as `ExtraJourney`) - and the unique id passed in `EstimatedVehicleJourneyCode` (ET) is used as the reference. There is no field similar to `EstimatedVehicleJourneyCode` in the `Affects`-object in SIRI.

Using `entityResolver.resolveId(...)` allows the SX-message to be processed before the actual ET-message is processed.

Reply:
In that case, but we should first try to resolve the DSJ from the DSJ id, and add the SJ id for the DSJ, only if it does not resolve, we should use the DSJ id as the SJ id. 

Now if we correctly give `XXX:DatedServiceJoureny:YYY-20230308`, which is part of `XXX:ServiceJoureny:YYY`, in `datedVehicleJourneyReves`, we will add that to `tripIds`, and not `XXX:ServiceJoureny:YYY`.

Reply:
Now resolving DSJ first, and using `tripId` + `serviceDate` from TripOnServiceDate if found.